src/lib/ledger_catchup/ledger_catchup.ml

```diff
@@ -4,6 +4,27 @@ open Protocols.Coda_transition_frontier
 open Pipe_lib
 open Coda_base
 
+(** [Ledger_catchup] is a process that connects a foreign external transition
+    into a transition frontier by asking it's peer a path of external_transitions.
+    It receives the external_transition from [Catchup_monitor]. With that
+    external_transition, it will ask it's peers for a path of
+    external_transitions from it's root to the transition it is asking for. It
+    will then do some verifications on these breadcrumbs. If all the
+    breadcrumbs are valid, it will then send them to the [Processor] via
+    writing them to catchup_breadcrumbs_writer. Otherwise, the sender who send
+    these breadcrumbs will be punished. Here are the validations that are done
+    with receiving a trail of external_transitions:
+
+    1. The root should exist in the frontier. The frontier should not be
+    missing too many external_transitions, so the querying node should have the
+    root in it's transition_frontier.
+
+    2. Each transitions are checked through [Transition_processor.Validator]
+    and [Protocol_state_validator]
+
+    3. Verifying each staged_ledger_diff by iteratively folding through each
```

Comment:
I think some of this info changes with my type level validation branch, but I will update this when I merge that in.